controllers/virtualmachinewebconsolerequest/v1alpha2/webconsolerequest_controller.go

```diff
@@ -220,3 +214,40 @@ func (r *Reconciler) ReconcileOwnerReferences(ctx *pkgctx.WebConsoleRequestConte
 	ctx.WebConsoleRequest.SetOwnerReferences([]metav1.OwnerReference{ownerRef})
 	return nil
 }
+
+// ProxyAddress first attempts to get the proxy address through the API Server DNS Names.
+// If that is unset, though, fall back to using the virtual IP.
+func (r *Reconciler) ProxyAddress(ctx *pkgctx.WebConsoleRequestContextV1) (string, error) {
+	if !pkgcfg.FromContext(ctx).Features.SimplifiedEnablement {
+		return r.ProxyAddressFromVirtualIP(ctx)
+	}
+
+	// Attempt to use the API Server DNS Names to get the proxy address.
+	proxyAddress, err := webconsoleurl.ProxyServiceDNSName(ctx, r)
+
+	if err != nil {
+		return "", fmt.Errorf("failed to get proxy service URL: %w", err)
+	}
+
+	// If no API Server DNS Name exists, fall back to using the virtual IP.
+	if len(proxyAddress) == 0 {
+		return r.ProxyAddressFromVirtualIP(ctx)
+	}
+
+	return proxyAddress, nil
+}
+
+// ProxyAddressFromVirtualIP retrieves the virtual IP, which will be used as the Proxy Address.
+func (r *Reconciler) ProxyAddressFromVirtualIP(ctx *pkgctx.WebConsoleRequestContextV1) (string, error) {
```

Comment:
wondering if this method can also go into the pkg/util/webconsole/ now? @dilyar85 ?

Reply:
The thing is that the contexts for this and the v1alpha1 are different data types. Does this matter? 

If we move this to the webconsole directory, could we just use context.Context as the type?

Reply:
Moved it for now.

Reply:
Yep agree better to move to the util package (also commented on it below).

Reply:
Done.